c++/src/capnp/layout.c++

```diff
@@ -790,7 +790,7 @@ struct WireHelpers {
     switch (ref->kind()) {
       case WirePointer::STRUCT: {
         KJ_REQUIRE(boundsCheck(segment, ptr, ref->structRef.wordSize()),
-                   "Message contained out-of-bounds struct pointer.") {
+                   "Message contained out-of-bounds struct pointer. Is your memory corrupted?") {
```

Comment:
I think we could expand this out a bit...

```suggestion
                   "Message contained out-of-bounds struct pointer. This usually indicates that "
                   "the input data was corrupted, used a different encoding that specified (e.g. "
                   "packed vs. non-packed), or was not a Cap'n Proto message to begin with. Note "
                   "that this error is NOT due to a schema mismatch; the input is invalid "
                   "regardless of schema.") {
```

Perhaps all the text after the first sentence should be factored out into a macro.